src/inmanta/data/__init__.py

```diff
@@ -5110,6 +5110,8 @@ async def get_resources_since_version_raw(
                 resource: dict[str, object] = dict(raw_resource)
                 if "attributes" in resource:
                     resource["attributes"] = json.loads(resource["attributes"])
+                if projection is not None:
+                    assert set(projection) <= resource.keys()
```

Comment:
For [this review comment](https://github.com/inmanta/inmanta-core/pull/8545#discussion_r1900838177)

> Should be add check to verify that resource_id and attributes are present in projection if not None.